lib/bosh_deployment_resource/bosh.rb

```diff
@@ -3,11 +3,23 @@
 
 module BoshDeploymentResource
   class Bosh
-    def initialize(target, username, password, command_runner=CommandRunner.new)
+    def initialize(target, username, password, cert=nil, command_runner=CommandRunner.new)
       @target = target
       @username = username
       @password = password
+      @cert = cert
       @command_runner = command_runner
+
+      if cert && File.exist?(cert)
+        # Assumes UAA with SSL cert...
+        run("bosh -n target #{target} --ca-cert #{cert}")
+
+        # TODO: Fix "stty: standard input: Inappropriate ioctl for device" errors
+        run(
+          "echo -n \"$BOSH_USER\n$BOSH_PASSWORD\" | bosh login 1>/dev/null",
```

Comment:
Is it required to run `login` after `target`? Is there something you could use like `status` that you could really throw away the output without echoing user/pass as input, especially since you are setting them in the environment to be picked up by the bosh cli?